src/hotspot/cpu/aarch64/vm_version_aarch64.hpp

```diff
@@ -119,6 +119,7 @@ class VM_Version : public Abstract_VM_Version {
   static int dcache_line_size() {
     return (1 << ((_psr_info.ctr_el0 >> 16) & 0x0f)) * 4;
   }
+  static bool supports_on_spin_wait() { return true; }
```

Comment:
If this method unconditionally returns true, why do we need it (and the check in `Matcher::match_rule_supported(int opcode)` at all?

The other question is if the ISB instruction which is used to implement the OnSpinWait node is really available on ALL 64-bit ARM implementations. `VMVersion::supports...()` is intended for checking specific CPU features. If ISB is an instruction which isn't available on all 64-bit ARM implementations than we should check for it here and rename the function to something like `supports_isb()`. Otherwise I think this function can be removed.